File: src/main/java/com/marsraver/WledFx/wled/model/WledConfig.java
package com.marsraver.WledFx.wled.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class WledConfig {
    private Object rev;
    private long vid;
    private Object id;
    private Object nw;
    private Object ap;
    private Object wifi;
    private Object hw;
    private Object light;
    private Object def;
    @JsonProperty("if")
    private InterfaceConfig iface;
    private Object ol;
    private Object timers;
    private Object ota;
    private Object um;
    
    @Data
    public static class InterfaceConfig {
        private Object sync;
        private Object nodes;
        private LiveConfig live;
        private Object va;
        private Object mqtt;
        private Object hue;
        private Object ntp;
    }
    
    @Data
    public static class LiveConfig {
        private boolean en; // Live enabled
        private boolean mso;
        private boolean rlm;
        private int port; // Port
        private boolean mc; // Multicast
        private DmxConfig dmx; // DMX configuration
        private int timeout;
        private boolean maxbri;
        @JsonProperty("no-gc")
        private boolean noGc;
        private int offset;
    }
    
    @Data
    public static class DmxConfig {
        private int uni; // Universe
        private boolean seqskip; // Skip out-of-sequence packets
        private int e131prio; // E1.31 priority
        private int addr; // DMX start address
        private int dss; // DMX single LED timeout
        private int mode; // DMX mode (4 = Multi RGB)
    }
}

File: src/main/kotlin/com/marsraver/wledfx/wled/model/WledConfig.kt
package com.marsraver.wledfx.wled.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class WledConfig(
    var rev: Any? = null,
    var vid: Long = 0,
    var id: Any? = null,
    var nw: Any? = null,
    var ap: Any? = null,
    var wifi: Any? = null,
    var hw: Any? = null,
    var light: Any? = null,
    var def: Any? = null,
    @JsonProperty("if")
    var iface: InterfaceConfig? = null,
    var ol: Any? = null,
    var timers: Any? = null,
    var ota: Any? = null,
    var um: Any? = null,
) {
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class InterfaceConfig(
        var sync: Any? = null,
        var nodes: Any? = null,
        var live: LiveConfig? = null,
        var va: Any? = null,
        var mqtt: Any? = null,
        var hue: Any? = null,
        var ntp: Any? = null,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class LiveConfig(
        var en: Boolean = false,
        var mso: Boolean = false,
        var rlm: Boolean = false,
        var port: Int = 0,
        var mc: Boolean = false,
        var dmx: DmxConfig? = null,
        var timeout: Int = 0,
        var maxbri: Boolean = false,
        @JsonProperty("no-gc")
        var noGc: Boolean = false,
        var offset: Int = 0,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class DmxConfig(
        var uni: Int = 0,
        var seqskip: Boolean = false,
        var e131prio: Int = 0,
        var addr: Int = 0,
        var dss: Int = 0,
        var mode: Int = 0,
    )
}

